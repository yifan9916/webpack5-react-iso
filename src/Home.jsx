import React, { useState } from 'react';
import Helmet from 'react-helmet';

import { useHomeContext } from './home-context';

import './index.scss';
import styles from './Home.module.scss';

const Home = () => {
  const appContext = useHomeContext();

  return (
    <>
      <Helmet>
        <title>Home</title>
      </Helmet>
      <div className={styles.home}>
        <h1>{appContext.title}</h1>
        <Section />
      </div>
    </>
  );
};

const Section = () => {
  const [count, setCount] = useState(0);

  const handleClick = () => {
    setCount((previousCount) => previousCount + 1);
  };

  return (
    <div className={styles.section}>
      <p>This is a counter: {count}</p>
      <button onClick={handleClick}>Increment</button>
    </div>
  );
};

export default Home;
