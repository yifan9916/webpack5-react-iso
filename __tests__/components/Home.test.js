import React from 'react';
import { render } from 'enzyme';

import Home from '../../src/Home';

describe('<Home />', () => {
  it('renders and matches snapshot', () => {
    const component = render(<Home />);

    expect(component).toMatchSnapshot();
  });
});
