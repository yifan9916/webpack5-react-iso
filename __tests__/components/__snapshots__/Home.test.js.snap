// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders and matches snapshot 1`] = `
<div
  class="home"
>
  <h1 />
  <div
    class="section"
  >
    <p>
      This is a counter: 0
    </p>
    <button>
      Increment
    </button>
  </div>
</div>
`;
